adafruit_portalbase/graphics.py

```diff
@@ -0,0 +1,164 @@
+    """Graphics Base Class for the Portal-style libraries.
+
+    :param display: An initialized display.
+    :param default_bg: The path to your default background image file or a hex color.
+                       Defaults to 0x000000.
+    :param bool auto_refresh: Automatically refresh the eInk after writing to displayio.
+                              Defaults to True.
+    :param debug: Turn on debug print outs. Defaults to False.
+
+    """
+
+    # pylint: disable=too-many-instance-attributes, too-many-locals, too-many-branches, too-many-statements
+    def __init__(self, display, *, default_bg=0x000000, debug=False):
+
+        self._debug = debug
+        self.display = display
+
+        if self._debug:
+            print("Init display")
+        self.splash = displayio.Group(max_size=15)
+        self._qr_group = None
+        if self._debug:
+            print("Init background")
+        self._bg_group = displayio.Group(max_size=1)
+        self._bg_file = None
+        self.splash.append(self._bg_group)
+        self.display.show(self.splash)
+
+        # set the default background
+        if default_bg is not None:
+            self.set_background(default_bg)
+
+        gc.collect()
+
+    def set_background(self, file_or_color, position=None):
+        """The background image to a bitmap file.
+
+        :param file_or_color: The filename of the chosen background image, or a hex color.
+        :param tuple position: Optional x and y coordinates to place the background at.
+
+        """
+        while self._bg_group:
+            self._bg_group.pop()
+
+        if not position:
+            position = (0, 0)  # default in top corner
+
+        if not file_or_color:
+            return  # we're done, no background desired
+        if self._bg_file:
+            self._bg_file.close()
+        if isinstance(file_or_color, str):  # its a filenme:
+            self._bg_file = open(file_or_color, "rb")
+            background = displayio.OnDiskBitmap(self._bg_file)
+            self._bg_sprite = displayio.TileGrid(
+                background,
+                pixel_shader=displayio.ColorConverter(),
+                x=position[0],
+                y=position[1],
+            )
+        elif isinstance(file_or_color, int):
+            # Make a background color fill
+            color_bitmap = displayio.Bitmap(self.display.width, self.display.height, 1)
+            color_palette = displayio.Palette(1)
+            color_palette[0] = file_or_color
+            self._bg_sprite = displayio.TileGrid(
+                color_bitmap,
+                pixel_shader=color_palette,
+                x=position[0],
+                y=position[1],
+            )
+        else:
+            raise RuntimeError("Unknown type of background")
+        self._bg_group.append(self._bg_sprite)
+        gc.collect()
+
+    def qrcode(
+        self, qr_data, *, qr_size=1, x=0, y=0, qr_color=0x000000
+    ):  # pylint: disable=invalid-name
```

Comment:
I'd love to see this split out into it's own widget like label. That way we can use it outside of portal.

Reply:
🤔  Please elaborate. I think I know what you mean, but want to be sure. Do you mean as part of an existing library or a maybe new library?

Reply:
I was thinking a new library or even part of the QRCode library. It'd be an object that represents the QRCode on screen. The constructor would take the parameters here and the object would inherit from Group so you could use and manipulate it like a normal group.